File: OctopusNode/src/main/java/com/octopus/node/service/SubJobUpdate.java
package com.octopus.node.service;

import com.octopus.node.core.JobType;

public class SubJobUpdate {
    final int id;
    final SubJobUpdateType subjobUpdateType;
    final JobType subjobType;
    public SubJobUpdate(final int id, final SubJobUpdateType subjobUpdateType, final JobType subjobType) {
        this.id = id;
        this.subjobUpdateType = subjobUpdateType;
        this.subjobType = subjobType;
    }

    public int getId() {
        return id;
    }

    public SubJobUpdateType getSubjobUpdateType() {
        return subjobUpdateType;
    }

    public JobType getSubjobType() {
        return subjobType;
    }
}

File: OctopusService/src/main/kotlin/com/octopus/transport/SubJobUpdate.kt
package com.octopus.transport

import com.fasterxml.jackson.annotation.JsonProperty
import com.octopus.core.enums.JobType

class SubJobUpdate(
        @JsonProperty val id: Int? = null,
        @JsonProperty val subjobUpdateType: SubJobUpdateType? = null,
        @JsonProperty val subjobType: JobType? = null
)